File: src/BetfairMarketAnalyzer/Services/ReportGeneratorService.cs
using System.Text;
using BetfairMarketAnalyzer.Models;

namespace BetfairMarketAnalyzer.Services;

public class ReportGeneratorService
{
    public string GenerateComprehensiveReport(AnalysisResult analysis, MarketData marketData)
    {
        var report = new StringBuilder();

        // Header
        GenerateHeader(report, analysis, marketData);
        
        // Market Information
        GenerateMarketInformation(report, marketData, analysis);
        
        // Favorite Selection Analysis
        GenerateFavoriteAnalysis(report, analysis);
        
        // Technical Analysis Section
        GenerateTechnicalAnalysis(report, analysis);
        
        // Volume and Flow Analysis
        GenerateVolumeAnalysis(report, analysis);
        
        // Support/Resistance Analysis
        GenerateSupportResistanceAnalysis(report, analysis);
        
        // Trading Recommendations
        GenerateTradingRecommendations(report, analysis);
        
        // Risk Assessment
        GenerateRiskAssessment(report, analysis);
        
        // Summary and Key Findings
        GenerateSummary(report, analysis);

        return report.ToString();
    }

    private void GenerateHeader(StringBuilder report, AnalysisResult analysis, MarketData marketData)
    {
        report.AppendLine("# Betfair Market Analysis Report");
        report.AppendLine("## Generated using Skender.Stock.Indicators Technical Analysis");
        report.AppendLine();
        report.AppendLine($"**Analysis Time**: {analysis.AnalysisTime:yyyy-MM-dd HH:mm:ss}");
        report.AppendLine($"**Time to Race Start**: {analysis.TimeToRaceStart}");
        report.AppendLine();
    }

    private void GenerateMarketInformation(StringBuilder report, MarketData marketData, AnalysisResult analysis)
    {
        report.AppendLine("## Market Information");
        report.AppendLine("```");
        report.AppendLine($"Market: {marketData.MarketName}");
        report.AppendLine($"Event: {marketData.EventName}");
        report.AppendLine($"Market ID: {marketData.MarketId}");
        report.AppendLine($"Status: {marketData.Status}");
        report.AppendLine($"Event Type: {marketData.EventType}");
        report.AppendLine($"Start Time: {marketData.StartTime:yyyy-MM-dd HH:mm:ss}");
        report.AppendLine($"Analysis Time: {analysis.AnalysisTime:yyyy-MM-dd HH:mm:ss}");
        report.AppendLine("```");
        report.AppendLine();
    }

    private void GenerateFavoriteAnalysis(StringBuilder report, AnalysisResult analysis)
    {
        report.AppendLine("## Favorite Selection Analysis Report");
        report.AppendLine();
        
        report.AppendLine("### Selection Overview");
        report.AppendLine($"- **Selection Name**: {analysis.SelectionName}");
        report.AppendLine($"- **Current Odds**: {analysis.CurrentOdds:F2}");
        report.AppendLine($"- **Implied Probability**: {analysis.ImpliedProbability:F1}% (P = 1/{analysis.CurrentOdds:F2})");
        report.AppendLine($"- **Opening Odds**: {analysis.OpeningOdds:F2}");
        report.AppendLine($"- **Odds Movement**: {analysis.OddsMovement:+0.00;-0.00} points ({analysis.OddsMovementPercentage:+0.0;-0.0}% change)");
        report.AppendLine($"- **Position in Market**: Market favorite with lowest odds");
        report.AppendLine();

        var movementDescription = analysis.OddsMovement < 0 ? "significant shortening (backing pressure)" : 
                                 analysis.OddsMovement > 0 ? "lengthening (laying pressure)" : "stable";
        report.AppendLine($"- **Movement Analysis**: {movementDescription}");
        report.AppendLine();
    }

    private void GenerateTechnicalAnalysis(StringBuilder report, AnalysisResult analysis)
    {
        report.AppendLine("### Technical Analysis (Skender.Stock.Indicators)");
        report.AppendLine();
        
        report.AppendLine("#### Technical Indicators");
        report.AppendLine($"- **RSI (14-period)**: {analysis.Indicators.RSI?.ToString("F1") ?? "N/A"}");
        report.AppendLine($"- **SMA (5-period)**: {analysis.Indicators.SMA5?.ToString("F2") ?? "N/A"}");
        report.AppendLine($"- **SMA (10-period)**: {analysis.Indicators.SMA10?.ToString("F2") ?? "N/A"}");
        report.AppendLine($"- **EMA (5-period)**: {analysis.Indicators.EMA5?.ToString("F2") ?? "N/A"}");
        report.AppendLine($"- **EMA (10-period)**: {analysis.Indicators.EMA10?.ToString("F2") ?? "N/A"}");
        
        if (analysis.Indicators.MACD.HasValue)
        {
            report.AppendLine($"- **MACD**: {analysis.Indicators.MACD:F3}");
            report.AppendLine($"- **MACD Signal**: {analysis.Indicators.MACDSignal?.ToString("F3") ?? "N/A"}");
        }
        
        if (analysis.Indicators.BollingerUpper.HasValue)
        {
            report.AppendLine($"- **Bollinger Upper**: {analysis.Indicators.BollingerUpper:F2}");
            report.AppendLine($"- **Bollinger Middle**: {analysis.Indicators.BollingerMiddle?.ToString("F2") ?? "N/A"}");
            report.AppendLine($"- **Bollinger Lower**: {analysis.Indicators.BollingerLower?.ToString("F2") ?? "N/A"}");
        }
        
        report.AppendLine($"- **Momentum Direction**: {analysis.Indicators.MomentumDirection}");
        report.AppendLine($"- **Volume Moving Average**: {analysis.Indicators.VolumeMovingAverage:F1}");
        report.AppendLine();

        report.AppendLine("#### Trend Analysis");
        report.AppendLine($"- **Current Trend**: {analysis.TrendDirection}");
        report.AppendLine($"- **Total Trading Volume**: {analysis.TotalVolume:F1} units");
        
        // RSI interpretation
        if (analysis.Indicators.RSI.HasValue)
        {
            var rsiInterpretation = analysis.Indicators.RSI < 30 ? "Oversold (potential bounce)" :
                                   analysis.Indicators.RSI > 70 ? "Overbought (potential reversal)" :
                                   "Neutral";
            report.AppendLine($"- **RSI Interpretation**: {rsiInterpretation}");
        }
        
        report.AppendLine();
    }

    private void GenerateVolumeAnalysis(StringBuilder report, AnalysisResult analysis)
    {
        report.AppendLine("### Volume and Market Flow Analysis");
        report.AppendLine();
        
        report.AppendLine("#### Market Flow Indicators");
        report.AppendLine($"- **Dominant Direction**: {analysis.MarketFlow.DominantDirection}");
        report.AppendLine($"- **Backing Pressure**: {analysis.MarketFlow.BackingPressure:F1}%");
        report.AppendLine($"- **Laying Pressure**: {analysis.MarketFlow.LayingPressure:F1}%");
        report.AppendLine($"- **Market Sentiment**: {analysis.MarketFlow.MarketSentiment}");
        report.AppendLine($"- **Steam Move Detected**: {(analysis.MarketFlow.SteamMoveDetected ? "Yes" : "No")}");
        
        if (analysis.MarketFlow.LastSteamMove.HasValue)
        {
            report.AppendLine($"- **Last Steam Move**: {analysis.MarketFlow.LastSteamMove:HH:mm:ss}");
        }
        
        report.AppendLine($"- **Recent Volume Ratio**: {analysis.MarketFlow.RecentVolumeRatio:F2}");
        report.AppendLine();

        if (analysis.VolumeSpikes.Any())
        {
            report.AppendLine("#### Significant Volume Activity");
            foreach (var spike in analysis.VolumeSpikes.Take(5))
            {
                report.AppendLine($"- **{spike.Time:HH:mm:ss}** - {spike.Type} at {spike.Price:F2} " +
                                $"(Volume: {spike.Volume:F1}, Direction: {spike.Direction})");
            }
            report.AppendLine();
        }
    }

    private void GenerateSupportResistanceAnalysis(StringBuilder report, AnalysisResult analysis)
    {
        report.AppendLine("### Support and Resistance Analysis");
        report.AppendLine();
        
        if (analysis.SupportLevels.Any())
        {
            report.AppendLine("#### Key Support Levels");
            foreach (var support in analysis.SupportLevels.Take(3))
            {
                report.AppendLine($"- **{support.Level:F2}**: {support.Strength} support " +
                                $"({support.TouchCount} touches, Volume: {support.TotalVolume:F1})");
            }
            report.AppendLine();
        }

        if (analysis.ResistanceLevels.Any())
        {
            report.AppendLine("#### Key Resistance Levels");
            foreach (var resistance in analysis.ResistanceLevels.Take(3))
            {
                report.AppendLine($"- **{resistance.Level:F2}**: {resistance.Strength} resistance " +
                                $"({resistance.TouchCount} touches, Volume: {resistance.TotalVolume:F1})");
            }
            report.AppendLine();
        }
    }

    private void GenerateTradingRecommendations(StringBuilder report, AnalysisResult analysis)
    {
        report.AppendLine("## Trading Strategy Recommendations");
        report.AppendLine();

        if (!analysis.TradingRecommendations.Any())
        {
            report.AppendLine("No specific trading recommendations generated based on current market conditions.");
            return;
        }

        for (int i = 0; i < analysis.TradingRecommendations.Count; i++)
        {
            var rec = analysis.TradingRecommendations[i];
            
            report.AppendLine($"### Strategy {i + 1}: {rec.Strategy}");
            report.AppendLine();
            
            report.AppendLine($"- **Action**: {rec.Action}");
            report.AppendLine($"- **Entry Price**: {rec.EntryPrice:F2}");
            
            if (rec.TargetPrice > 0)
                report.AppendLine($"- **Target Price**: {rec.TargetPrice:F2}");
            else
                report.AppendLine($"- **Target**: Hold to win");
                
            report.AppendLine($"- **Stop Loss**: {rec.StopLoss:F2}");
            report.AppendLine($"- **Expected Profit**: {rec.ExpectedProfit:F1}%");
            report.AppendLine($"- **Risk/Reward Ratio**: {rec.RiskReward:F1}");
            report.AppendLine($"- **Time Frame**: {rec.TimeFrame}");
            report.AppendLine($"- **Risk Level**: {rec.RiskLevel}");
            report.AppendLine($"- **Confidence Level**: {rec.ConfidenceLevel}/10");
            report.AppendLine($"- **Position Sizing**: {rec.PositionSizing}% of bankroll");
            report.AppendLine($"- **Logic**: {rec.Logic}");
            report.AppendLine();
        }
    }

    private void GenerateRiskAssessment(StringBuilder report, AnalysisResult analysis)
    {
        report.AppendLine("### Risk Assessment and Position Sizing");
        report.AppendLine();
        
        var timeToRace = analysis.TimeToRaceStart.TotalMinutes;
        var riskLevel = timeToRace < 5 ? "High" : timeToRace < 15 ? "Moderate" : "Low";
        
        report.AppendLine($"- **Time Risk**: {riskLevel} (Race starts in {timeToRace:F0} minutes)");
        report.AppendLine($"- **Volatility Risk**: {(analysis.VolumeSpikes.Count > 5 ? "High" : "Moderate")} " +
                         $"({analysis.VolumeSpikes.Count} recent volume spikes)");
        
        var liquidityRisk = analysis.TotalVolume > 1000 ? "Low" : analysis.TotalVolume > 500 ? "Moderate" : "High";
        report.AppendLine($"- **Liquidity Risk**: {liquidityRisk} (Total volume: {analysis.TotalVolume:F0})");
        
        report.AppendLine();
        
        report.AppendLine("#### Recommended Position Sizing");
        report.AppendLine("- **Conservative**: 1-2% of bankroll");
        report.AppendLine("- **Moderate**: 3-5% of bankroll");
        report.AppendLine("- **Aggressive**: 5-10% of bankroll");
        report.AppendLine();
    }

    private void GenerateSummary(StringBuilder report, AnalysisResult analysis)
    {
        report.AppendLine("## Key Findings Summary");
        report.AppendLine();
        
        report.AppendLine($"- **Favorite Selection**: {analysis.SelectionName} at {analysis.CurrentOdds:F2} odds " +
                         $"({analysis.ImpliedProbability:F1}% implied probability)");
        report.AppendLine($"- **Odds Movement**: {analysis.OddsMovement:+0.00;-0.00} points from opening " +
                         $"({analysis.OddsMovementPercentage:+0.0;-0.0}% change)");
        report.AppendLine($"- **Current Trend**: {analysis.TrendDirection}");
        report.AppendLine($"- **Liquidity Status**: {(analysis.TotalVolume > 1000 ? "Excellent" : "Good")} " +
                         $"- {analysis.TotalVolume:F0} units traded");
        report.AppendLine($"- **Market Flow**: {analysis.MarketFlow.DominantDirection} pressure dominant " +
                         $"({analysis.MarketFlow.BackingPressure:F0}% backing vs {analysis.MarketFlow.LayingPressure:F0}% laying)");
        
        if (analysis.MarketFlow.SteamMoveDetected)
        {
            report.AppendLine($"- **Steam Move Alert**: Detected - significant informed money activity");
        }
        
        report.AppendLine();
        
        if (analysis.TradingRecommendations.Any())
        {
            var topRec = analysis.TradingRecommendations.First();
            report.AppendLine("### Top Recommendation");
            report.AppendLine($"- **Strategy**: {topRec.Strategy}");
            report.AppendLine($"- **Entry Price**: {topRec.EntryPrice:F2}");
            report.AppendLine($"- **Target**: {(topRec.TargetPrice > 0 ? topRec.TargetPrice.ToString("F2") : "Hold to win")}");
            report.AppendLine($"- **Risk Level**: {topRec.RiskLevel}");
            report.AppendLine($"- **Confidence**: {topRec.ConfidenceLevel}/10");
            report.AppendLine($"- **Expected Return**: {topRec.ExpectedProfit:F1}%");
        }
        
        report.AppendLine();
        report.AppendLine("---");
        report.AppendLine("*Analysis generated using Skender.Stock.Indicators library for technical analysis*");
        report.AppendLine($"*Report generated at: {DateTime.Now:yyyy-MM-dd HH:mm:ss}*");
    }
}

File: src/BetfairMarketAnalyzerPython/services/report_generator_service.py
from datetime import datetime, timedelta
from typing import List

from models.betfair_models import MarketData
from models.analysis_models import AnalysisResult


class ReportGeneratorService:
    
    def generate_comprehensive_report(self, analysis: AnalysisResult, market_data: MarketData) -> str:
        """
        Generate a comprehensive markdown report from analysis results
        """
        report_lines = []
        
        # Header
        self._generate_header(report_lines, analysis, market_data)
        
        # Market Information
        self._generate_market_information(report_lines, market_data, analysis)
        
        # Favorite Selection Analysis
        self._generate_favorite_analysis(report_lines, analysis)
        
        # Technical Analysis Section
        self._generate_technical_analysis(report_lines, analysis)
        
        # Volume and Flow Analysis
        self._generate_volume_analysis(report_lines, analysis)
        
        # Support/Resistance Analysis
        self._generate_support_resistance_analysis(report_lines, analysis)
        
        # Trading Recommendations
        self._generate_trading_recommendations(report_lines, analysis)
        
        # Risk Assessment
        self._generate_risk_assessment(report_lines, analysis)
        
        # Summary and Key Findings
        self._generate_summary(report_lines, analysis)
        
        return '\n'.join(report_lines)
    
    def _generate_header(self, report_lines: List[str], analysis: AnalysisResult, market_data: MarketData):
        """Generate report header"""
        report_lines.extend([
            "# Betfair Market Analysis Report",
            "## Generated using TA-Lib Technical Analysis (Python)",
            "",
            f"**Analysis Time**: {analysis.analysis_time.strftime('%Y-%m-%d %H:%M:%S')}",
            f"**Time to Race Start**: {analysis.time_to_race_start}",
            ""
        ])
    
    def _generate_market_information(self, report_lines: List[str], market_data: MarketData, analysis: AnalysisResult):
        """Generate market information section"""
        report_lines.extend([
            "## Market Information",
            "```",
            f"Market: {market_data.market_name}",
            f"Event: {market_data.event_name}",
            f"Market ID: {market_data.market_id}",
            f"Status: {market_data.status}",
            f"Event Type: {market_data.event_type}",
            f"Start Time: {market_data.start_time.strftime('%Y-%m-%d %H:%M:%S')}",
            f"Analysis Time: {analysis.analysis_time.strftime('%Y-%m-%d %H:%M:%S')}",
            "```",
            ""
        ])
    
    def _generate_favorite_analysis(self, report_lines: List[str], analysis: AnalysisResult):
        """Generate favorite selection analysis"""
        report_lines.extend([
            "## Favorite Selection Analysis Report",
            "",
            "### Selection Overview",
            f"- **Selection Name**: {analysis.selection_name}",
            f"- **Current Odds**: {analysis.current_odds:.2f}",
            f"- **Implied Probability**: {analysis.implied_probability:.1f}% (P = 1/{analysis.current_odds:.2f})",
            f"- **Opening Odds**: {analysis.opening_odds:.2f}",
            f"- **Odds Movement**: {analysis.odds_movement:+.2f} points ({analysis.odds_movement_percentage:+.1f}% change)",
            f"- **Position in Market**: Market favorite with lowest odds",
            ""
        ])
        
        movement_description = (
            "significant shortening (backing pressure)" if analysis.odds_movement < 0 else
            "lengthening (laying pressure)" if analysis.odds_movement > 0 else
            "stable"
        )
        report_lines.extend([
            f"- **Movement Analysis**: {movement_description}",
            ""
        ])
    
    def _generate_technical_analysis(self, report_lines: List[str], analysis: AnalysisResult):
        """Generate technical analysis section"""
        indicators = analysis.indicators
        
        report_lines.extend([
            "### Technical Analysis (TA-Lib)",
            "",
            "#### Technical Indicators",
            f"- **RSI (14-period)**: {f'{indicators.rsi:.1f}' if indicators.rsi is not None else 'N/A'}",
            f"- **SMA (5-period)**: {f'{indicators.sma5:.2f}' if indicators.sma5 is not None else 'N/A'}",
            f"- **SMA (10-period)**: {f'{indicators.sma10:.2f}' if indicators.sma10 is not None else 'N/A'}",
            f"- **EMA (5-period)**: {f'{indicators.ema5:.2f}' if indicators.ema5 is not None else 'N/A'}",
            f"- **EMA (10-period)**: {f'{indicators.ema10:.2f}' if indicators.ema10 is not None else 'N/A'}"
        ])
        
        if indicators.macd is not None:
            report_lines.extend([
                f"- **MACD**: {indicators.macd:.3f}",
                f"- **MACD Signal**: {f'{indicators.macd_signal:.3f}' if indicators.macd_signal is not None else 'N/A'}"
            ])
        
        if indicators.bollinger_upper is not None:
            report_lines.extend([
                f"- **Bollinger Upper**: {indicators.bollinger_upper:.2f}",
                f"- **Bollinger Middle**: {f'{indicators.bollinger_middle:.2f}' if indicators.bollinger_middle is not None else 'N/A'}",
                f"- **Bollinger Lower**: {f'{indicators.bollinger_lower:.2f}' if indicators.bollinger_lower is not None else 'N/A'}"
            ])
        
        report_lines.extend([
            f"- **Momentum Direction**: {indicators.momentum_direction}",
            f"- **Volume Moving Average**: {indicators.volume_moving_average:.1f}",
            "",
            "#### Trend Analysis",
            f"- **Current Trend**: {analysis.trend_direction}",
            f"- **Total Trading Volume**: {analysis.total_volume:.1f} units"
        ])
        
        # RSI interpretation
        if indicators.rsi is not None:
            rsi_interpretation = (
                "Oversold (potential bounce)" if indicators.rsi < 30 else
                "Overbought (potential reversal)" if indicators.rsi > 70 else
                "Neutral"
            )
            report_lines.append(f"- **RSI Interpretation**: {rsi_interpretation}")
        
        report_lines.append("")
    
    def _generate_volume_analysis(self, report_lines: List[str], analysis: AnalysisResult):
        """Generate volume and market flow analysis"""
        market_flow = analysis.market_flow
        
        report_lines.extend([
            "### Volume and Market Flow Analysis",
            "",
            "#### Market Flow Indicators",
            f"- **Dominant Direction**: {market_flow.dominant_direction}",
            f"- **Backing Pressure**: {market_flow.backing_pressure:.1f}%",
            f"- **Laying Pressure**: {market_flow.laying_pressure:.1f}%",
            f"- **Market Sentiment**: {market_flow.market_sentiment}",
            f"- **Steam Move Detected**: {'Yes' if market_flow.steam_move_detected else 'No'}"
        ])
        
        if market_flow.last_steam_move:
            report_lines.append(f"- **Last Steam Move**: {market_flow.last_steam_move.strftime('%H:%M:%S')}")
        
        report_lines.extend([
            f"- **Recent Volume Ratio**: {market_flow.recent_volume_ratio:.2f}",
            ""
        ])
        
        if analysis.volume_spikes:
            report_lines.extend([
                "#### Significant Volume Activity"
            ])
            for spike in analysis.volume_spikes[:5]:
                report_lines.append(
                    f"- **{spike.time.strftime('%H:%M:%S')}** - {spike.type} at {spike.price:.2f} "
                    f"(Volume: {spike.volume:.1f}, Direction: {spike.direction})"
                )
            report_lines.append("")
    
    def _generate_support_resistance_analysis(self, report_lines: List[str], analysis: AnalysisResult):
        """Generate support and resistance analysis"""
        report_lines.extend([
            "### Support and Resistance Analysis",
            ""
        ])
        
        if analysis.support_levels:
            report_lines.append("#### Key Support Levels")
            for support in analysis.support_levels[:3]:
                report_lines.append(
                    f"- **{support.level:.2f}**: {support.strength} support "
                    f"({support.touch_count} touches, Volume: {support.total_volume:.1f})"
                )
            report_lines.append("")
        
        if analysis.resistance_levels:
            report_lines.append("#### Key Resistance Levels")
            for resistance in analysis.resistance_levels[:3]:
                report_lines.append(
                    f"- **{resistance.level:.2f}**: {resistance.strength} resistance "
                    f"({resistance.touch_count} touches, Volume: {resistance.total_volume:.1f})"
                )
            report_lines.append("")
    
    def _generate_trading_recommendations(self, report_lines: List[str], analysis: AnalysisResult):
        """Generate trading recommendations section"""
        report_lines.extend([
            "## Trading Strategy Recommendations",
            ""
        ])
        
        if not analysis.trading_recommendations:
            report_lines.append("No specific trading recommendations generated based on current market conditions.")
            return
        
        for i, rec in enumerate(analysis.trading_recommendations, 1):
            report_lines.extend([
                f"### Strategy {i}: {rec.strategy}",
                "",
                f"- **Action**: {rec.action}",
                f"- **Entry Price**: {rec.entry_price:.2f}"
            ])
            
            if rec.target_price > 0:
                report_lines.append(f"- **Target Price**: {rec.target_price:.2f}")
            else:
                report_lines.append("- **Target**: Hold to win")
            
            report_lines.extend([
                f"- **Stop Loss**: {rec.stop_loss:.2f}",
                f"- **Expected Profit**: {rec.expected_profit:.1f}%",
                f"- **Risk/Reward Ratio**: {rec.risk_reward:.1f}",
                f"- **Time Frame**: {rec.time_frame}",
                f"- **Risk Level**: {rec.risk_level}",
                f"- **Confidence Level**: {rec.confidence_level}/10",
                f"- **Position Sizing**: {rec.position_sizing}% of bankroll",
                f"- **Logic**: {rec.logic}",
                ""
            ])
    
    def _generate_risk_assessment(self, report_lines: List[str], analysis: AnalysisResult):
        """Generate risk assessment section"""
        time_to_race = analysis.time_to_race_start.total_seconds() / 60  # Convert to minutes
        risk_level = "High" if time_to_race < 5 else "Moderate" if time_to_race < 15 else "Low"
        
        report_lines.extend([
            "### Risk Assessment and Position Sizing",
            "",
            f"- **Time Risk**: {risk_level} (Race starts in {time_to_race:.0f} minutes)",
            f"- **Volatility Risk**: {'High' if len(analysis.volume_spikes) > 5 else 'Moderate'} "
            f"({len(analysis.volume_spikes)} recent volume spikes)"
        ])
        
        liquidity_risk = (
            "Low" if analysis.total_volume > 1000 else
            "Moderate" if analysis.total_volume > 500 else
            "High"
        )
        report_lines.extend([
            f"- **Liquidity Risk**: {liquidity_risk} (Total volume: {analysis.total_volume:.0f})",
            "",
            "#### Recommended Position Sizing",
            "- **Conservative**: 1-2% of bankroll",
            "- **Moderate**: 3-5% of bankroll",
            "- **Aggressive**: 5-10% of bankroll",
            ""
        ])
    
    def _generate_summary(self, report_lines: List[str], analysis: AnalysisResult):
        """Generate summary section"""
        report_lines.extend([
            "## Key Findings Summary",
            "",
            f"- **Favorite Selection**: {analysis.selection_name} at {analysis.current_odds:.2f} odds "
            f"({analysis.implied_probability:.1f}% implied probability)",
            f"- **Odds Movement**: {analysis.odds_movement:+.2f} points from opening "
            f"({analysis.odds_movement_percentage:+.1f}% change)",
            f"- **Current Trend**: {analysis.trend_direction}",
            f"- **Liquidity Status**: {'Excellent' if analysis.total_volume > 1000 else 'Good'} "
            f"- {analysis.total_volume:.0f} units traded",
            f"- **Market Flow**: {analysis.market_flow.dominant_direction} pressure dominant "
            f"({analysis.market_flow.backing_pressure:.0f}% backing vs {analysis.market_flow.laying_pressure:.0f}% laying)"
        ])
        
        if analysis.market_flow.steam_move_detected:
            report_lines.append("- **Steam Move Alert**: Detected - significant informed money activity")
        
        report_lines.append("")
        
        if analysis.trading_recommendations:
            top_rec = analysis.trading_recommendations[0]
            report_lines.extend([
                "### Top Recommendation",
                f"- **Strategy**: {top_rec.strategy}",
                f"- **Entry Price**: {top_rec.entry_price:.2f}",
                f"- **Target**: {top_rec.target_price:.2f if top_rec.target_price > 0 else 'Hold to win'}",
                f"- **Risk Level**: {top_rec.risk_level}",
                f"- **Confidence**: {top_rec.confidence_level}/10",
                f"- **Expected Return**: {top_rec.expected_profit:.1f}%"
            ])
        
        report_lines.extend([
            "",
            "---",
            "*Analysis generated using TA-Lib library for technical analysis*",
            f"*Report generated at: {datetime.now().strftime('%Y-%m-%d %H:%M:%S')}*"
        ])
